src/cts/src/TritonCTS.cpp

```diff
@@ -1333,4 +1340,190 @@ double TritonCTS::computeInsertionDelay(const std::string& name,
+  // first, look for inverters with "is_clock_cell: true" cell attribute
+  std::vector<sta::LibertyCell*> inverters;
+  sta::LibertyLibraryIterator* lib_iter = network_->libertyLibraryIterator();
+  while (lib_iter->hasNext()) {
+    sta::LibertyLibrary* lib = lib_iter->next();
+    for (sta::LibertyCell* inv : *lib->inverters()) {
+      if (inv->isClockCell() && isClockCellCandidate(inv)) {
+        inverters.emplace_back(inv);
+        dummyCandidates.emplace_back(inv);
+      }
+    }
+  }
+
+  // second, look for all inverters with name CLKINV or clkinv
+  if (inverters.empty()) {
+    sta::PatternMatch patternClkInv("*CLKINV*",
+                                    /* is_regexp */ true,
+                                    /* nocase */ true,
+                                    /* Tcl_interp* */ nullptr);
+    lib_iter = network_->libertyLibraryIterator();
+    while (lib_iter->hasNext()) {
+      sta::LibertyLibrary* lib = lib_iter->next();
+      for (sta::LibertyCell* inv :
+           lib->findLibertyCellsMatching(&patternClkInv)) {
+        if (inv->isInverter() && isClockCellCandidate(inv)) {
+          inverters.emplace_back(inv);
+          dummyCandidates.emplace_back(inv);
+        }
+      }
+    }
+  }
+
+  // third, look for all inverters with name INV or inv
+  if (inverters.empty()) {
+    sta::PatternMatch patternInv("*INV*",
+                                 /* is_regexp */ true,
+                                 /* nocase */ true,
+                                 /* Tcl_interp* */ nullptr);
+    lib_iter = network_->libertyLibraryIterator();
+    while (lib_iter->hasNext()) {
+      sta::LibertyLibrary* lib = lib_iter->next();
+      for (sta::LibertyCell* inv : lib->findLibertyCellsMatching(&patternInv)) {
+        if (inv->isInverter() && isClockCellCandidate(inv)) {
+          inverters.emplace_back(inv);
+          dummyCandidates.emplace_back(inv);
+        }
+      }
+    }
+  }
+
+  // abandon attributes & name patterns, just look for all inverters
+  if (inverters.empty()) {
+    lib_iter = network_->libertyLibraryIterator();
+    while (lib_iter->hasNext()) {
+      sta::LibertyLibrary* lib = lib_iter->next();
+      for (sta::LibertyCell* inv : *lib->inverters()) {
+        if (isClockCellCandidate(inv)) {
+          inverters.emplace_back(inv);
+          dummyCandidates.emplace_back(inv);
+        }
+      }
+    }
+  }
+
+  // Sort cells in ascending order of input cap
+  std::sort(dummyCandidates.begin(),
+            dummyCandidates.end(),
+            [](const sta::LibertyCell* cell1, const sta::LibertyCell* cell2) {
+              return (getInputCap(cell1) < getInputCap(cell2));
+            });
+
+  if (logger_->debugCheck(utl::CTS, "dummy load", 1)) {
+    for (const sta::LibertyCell* libCell : dummyCandidates) {
+      logger_->report("{} is a dummy cell candidate with input cap={:0.3e}",
+                      libCell->name(),
+                      getInputCap(libCell));
```

Comment:
you should be calling debug() not report for debug output

Reply:
Done.